### ISSUE TITLE
Add full node vs archive node attribute

### ISSUE THREAD

**Issue by 0xprames** [Keep-Reth-Strange/reth-crawler#53]
https://x.com/pistomat_eth/status/1721551057724047809?s=20

**Comment by 0xprames** [comment]
this needs some investigation - looking into this

**Comment by 0xprames** [comment]
i think #14 may help here - but not sure, ive unassigned myself for now to leave this open for @alessandromazza98 to explore!

**Comment by alessandromazza98** [comment]
Yes, I can take it after https://github.com/Keep-Reth-Strange/reth-crawler/issues/67

**Comment by alessandromazza98** [comment]
I'm looking right now into research for it.

Up to now my idea is just to use an RPC call to ask for some old state data with rpc calls such as:

- `eth_getTransactionCount`
- `eth_getBalance`

and some others.

> this approach assumes:
> 
> * node's RPC (8545) is open for external use, which isn't always the case
> * the query doesn't fail for any other reasons
> * the historical state data is "legitimate" (not sybilled, spoofed etc)

It is the same approach. I actually don't like it a lot if I have to be honest because there are too many scenarios in which a node does not answer the rpc call...

But yeah, it's a first step

**Comment by alessandromazza98** [comment]
Testing the RPC approach... It doesn't look good right now.

You can have a look in this branch: https://github.com/Keep-Reth-Strange/reth-crawler/tree/ale/full-archive-nodes

I'm just testing with a `eth_getBalance` RPC call, asking for the balance of a random address (that has balance != 0) at 10_000_000 block

The problem I'm facing is that all (at least for now) peers doesn't have the RPC 8545 port opened and so I'm not able to get the information if they are archive or full.
